app/src/main/java/com/dcrandroid/activities/ProposalDetailsActivity.kt

```diff
@@ -0,0 +1,179 @@
+    private lateinit var proposalDescription: TextView
+    private lateinit var votePercent: TextView
+    private lateinit var author: TextView
+    private lateinit var timestamp: TextView
+    private lateinit var comments: TextView
+    private lateinit var version: TextView
+    private lateinit var status: TextView
+    private lateinit var yes: TextView
+    private lateinit var no: TextView
+    private lateinit var progressBar: ProgressBar
+    private lateinit var voteProgress: ProgressBar
+    private val gson = GsonBuilder().registerTypeHierarchyAdapter(ArrayList::class.java, Deserializer.ProposalDeserializer()).create()
+    private var proposal: Proposal? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_proposal_details)
+
+        proposalTitle = findViewById(R.id.proposal_title)
+        proposalDescription = findViewById(R.id.proposal_description)
+        votePercent = findViewById(R.id.vote_progress)
+        author = findViewById(R.id.proposal_author)
+        timestamp = findViewById(R.id.proposal_timestamp)
+        comments = findViewById(R.id.proposal_comments)
+        version = findViewById(R.id.proposal_version)
+        progressBar = findViewById(R.id.progress)
+        voteProgress = findViewById(R.id.progressBar)
+        status = findViewById(R.id.proposal_status)
+        yes = findViewById(R.id.yes_votes)
+        no = findViewById(R.id.no_votes)
+
+        proposal = intent.getSerializableExtra("proposal") as Proposal
+
+        loadProposalDetails()
+
+        open_proposal.setOnClickListener {
+            val url = "http://proposals.decred.org/proposals/" + proposal!!.censorshipRecord!!.token
+            val i = Intent(Intent.ACTION_VIEW)
+            i.data = Uri.parse(url)
+            startActivity(i)
+        }
+
+        share_proposal.setOnClickListener {
+            val share = Intent(Intent.ACTION_SEND)
+            share.type = "text/plain"
+            share.addFlags(Intent.FLAG_ACTIVITY_NEW_DOCUMENT)
+            share.putExtra(Intent.EXTRA_SUBJECT, proposal!!.name)
+            share.putExtra(Intent.EXTRA_TEXT, "http://proposals.decred.org/proposals/" + proposal!!.censorshipRecord!!.token)
+            startActivity(Intent.createChooser(share, "Share Proposal Link"))
+        }
+
+        go_back.setOnClickListener {
+            finish()
+        }
+    }
+
+    private fun loadProposalDetails() = GlobalScope.launch(Dispatchers.Default) {
+        val proposalResult = multiWallet!!.politeia!!.getProposalByID(proposal!!.id)
+        val proposalObjectJson = JSONObject(proposalResult).getJSONObject("result")
+        val proposalResultString: String = proposalObjectJson.toString()
+
+        val voteObjectJson = proposalObjectJson.getJSONObject("votesummary")
+        val voteSummaryString: String = voteObjectJson.toString()
+
+        runOnUiThread {
```

Comment:
Most of the work done in this UI thread can be done in the background and the result can be posted to the UI.

Reply:
what do you mean done in the background?